src/Servers/HttpSys/src/RequestProcessing/RequestHeaders.Generated.tt

```diff
@@ -213,5 +213,36 @@ namespace Microsoft.AspNetCore.HttpSys.Internal
             }
 <# } #>
         }
+
+        private ReadOnlySpan<byte> HeaderKeys => new byte[]
+        {
+<# foreach(var prop in props) { #>
+            (byte)<#=prop.ID#>,
+<# } #>
+        };
+
+        private string GetHeaderKeyName(HttpSysRequestHeader header) => header switch
+        {
+<# foreach(var prop in props) { #>
+            <#=prop.ID#> => HeaderNames.<#=prop.Name#>,
+<# } #>
+            _ => header.ToString()
+        };
+
+        private bool IsOverridenKey(HttpSysRequestHeader header) => header switch
+        {
+<# foreach(var prop in props) { #>
+            <#=prop.ID#> => <#=IsRead(prop.Index)#>,
+<# } #>
+            _ => false,
+        };
+
+        private bool HasValue(HttpSysRequestHeader header) => header switch
+        {
+<# foreach(var prop in props) { #>
+            <#=prop.ID#> => _<#=prop.Name#> != StringValues.Empty,
```

Comment:
You should move some of the other checks in here. E.g. 
https://github.com/dotnet/aspnetcore/blob/ceffcc9f6db99bab0c53785912411706988410db/src/Servers/HttpSys/src/RequestProcessing/RequestHeaders.Generated.tt#L171-L172

Then PropertiesTryRemove could call this API too.

IsOverridenKey wouldn't be needed if that check was consolidated in here.

Reply:
very good point. Is checking the Count better to comparing against `StringValues.Empty`?

Reply:
> Is checking the Count `<#=prop.Name#>.Count > 0` better to comparing against `StringValues.Empty`?

They should be functionally the same, and 0 is shorter 😁

Reply:
I was considering perf, I would _assume_ `StringValues.Empty`, I will measure it.

Reply:
https://github.com/dotnet/runtime/blob/4deccc02588990626180eaa812e63603d6c2790e/src/libraries/Microsoft.Extensions.Primitives/src/StringValues.cs#L526-L530

Reply:
Ok, Count it is :)